xwayland/selection/incoming.c

```diff
@@ -492,8 +492,16 @@ int xwm_handle_xfixes_selection_notify(struct wlr_xwm *xwm,
 		return 0;
 	}
 
+	if (event->timestamp <= selection->timestamp) {
```

Comment:
I'm not sure this block is strictly necessary (never fired in my testing), but I don't think it can hurt.

Reply:
No prior art here; KWin doesn't do this so their selections can go "back in time": https://invent.kde.org/plasma/kwin/-/blob/master/src/xwl/selection.cpp#L106